src/nfts/NftGallery.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useEffect } from 'react'
+import { useTranslation } from 'react-i18next'
+import { FlatList, RefreshControl, StyleSheet, Text, View } from 'react-native'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import { useDispatch, useSelector } from 'react-redux'
+import { NftEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import Touchable from 'src/components/Touchable'
+import ImageErrorIcon from 'src/icons/ImageErrorIcon'
+import DrawerTopBar from 'src/navigator/DrawerTopBar'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import NftImage from 'src/nfts/NftImage'
+import NftsLoadError from 'src/nfts/NftsLoadError'
+import { nftsErrorSelector, nftsLoadingSelector, nftsSelector } from 'src/nfts/selectors'
+import { fetchNfts } from 'src/nfts/slice'
+import { NftOrigin } from 'src/nfts/types'
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import variables from 'src/styles/variables'
+
+// This is used instead of a gap between the images because the gap is not supported in RN yet.
+const imageSize = (variables.width - Spacing.Regular16 * 3) / 2
+
+export default function NftGallery() {
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+  const loading = useSelector(nftsLoadingSelector)
+  const error = useSelector(nftsErrorSelector)
+  const nfts = useSelector(nftsSelector)
+
+  useEffect(() => {
+    ValoraAnalytics.track(NftEvents.nft_gallery_screen_open)
+    dispatch(fetchNfts())
+  }, [])
+
+  return (
+    <SafeAreaView testID="NftGallery" style={styles.safeAreaContainer} edges={['top']}>
+      <DrawerTopBar middleElement={<Text>{t('nftGallery.title')}</Text>} />
+      {error ? (
+        <NftsLoadError testID="NftGallery/NftsLoadErrorScreen" />
+      ) : (
+        <FlatList
+          numColumns={2}
+          data={nfts}
+          refreshing={loading}
+          onRefresh={() => dispatch(fetchNfts())}
+          contentContainerStyle={styles.contentContainer}
+          refreshControl={
+            <RefreshControl
+              refreshing={loading}
+              colors={[colors.greenUI]}
+              onRefresh={() => dispatch(fetchNfts())}
+            />
+          }
+          renderItem={({ item, index }) => (
+            <Touchable
+              borderless={false}
+              onPress={() => navigate(Screens.NftsInfoCarousel, { nfts: [item] })}
+              style={[
+                styles.touchableIcon,
+                index % 2 === 0
+                  ? { marginRight: Spacing.Regular16 } &&
+                    index === nfts.length - 1 &&
+                    styles.lastImage
+                  : { marginLeft: Spacing.Regular16 },
```

Comment:
i found this a `{ marginRight: Spacing.Regular16 } && index === nfts.length - 1 && styles.lastImage`  a little unexpected for a style. can we simplify? don't we only need to remove the right margin when index % 2 === 0?

Reply:
Added a comment in 5a51564adb3e69c804800a184a43b2592b4d21ab to help explain why this was done.